src/pages/workspaces/workspace/Data.js

```diff
@@ -337,17 +343,44 @@ const EntitiesContent = ajaxCaller(class EntitiesContent extends Component {
         .paginatedEntitiesOfType(entityKey, {
           page: pageNumber, pageSize: itemsPerPage, sortField: sort.field, sortDirection: sort.direction
         })
-      this.setState({ entities: results, totalRowCount: unfilteredCount, selectedEntities: [] })
+      this.setState({ entities: results, totalRowCount: unfilteredCount })
+    } catch (error) {
+      reportError('Error loading entities', error)
+    } finally {
+      this.setState({ loading: false })
+    }
+  }
+
+  async selectAll() {
+    const { entityKey, workspace: { workspace: { namespace, name } }, ajax: { Workspaces } } = this.props
+    try {
+      this.setState({ loading: true })
+      const results = await Workspaces.workspace(namespace, name).entitiesOfType(entityKey)
+      this.setState({ selectedEntities: entityMap(results) })
     } catch (error) {
       reportError('Error loading entities', error)
     } finally {
       this.setState({ loading: false })
     }
   }
 
+  selectPage() {
+    const { entities, selectedEntities } = this.state
+    this.setState({ selectedEntities: _.assign(selectedEntities, entityMap(entities)) })
+  }
+
+  deselectPage() {
+    const { entities, selectedEntities } = this.state
+    this.setState({ selectedEntities: _.omitBy((v, k) => _.has(k, entityMap(entities)), selectedEntities) })
```

Comment:
Gotcha, something like `_.omit(_.map(a => [a], _.keys(entities)), selectedEntities)`.

@panentheos, I just noticed that this is another function that operates on paths rather than keys which is the reason for `_.map(a => [a])`.